packages/material-ui/src/styles/useThemeProps.d.ts

```diff
@@ -16,7 +12,7 @@ export interface AdditionalThemeProps<Theme> {
 }
 
 export default function useThemeProps<
-  Theme extends ThemeWithProps,
+  Theme extends MuiTheme,
```

Comment:
What are these changes and the ones above for?

Reply:
the `theme` return from `useThemeProps` has the type `ThemeWithProps` which is not the same `theme` as in experimentalStyled, so got type error. This will occur in any component with .tsx